x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -849,14 +869,91 @@ private DataType[] allowedEnrichTypes(String matchType) {
 
     private static List<Attribute> resolveAgainstList(UnresolvedNamePattern up, Collection<Attribute> attrList) {
         UnresolvedAttribute ua = new UnresolvedAttribute(up.source(), up.pattern());
-        Predicate<Attribute> matcher = a -> up.match(a.name());
-        var matches = AnalyzerRules.maybeResolveAgainstList(matcher, () -> ua, attrList, true, a -> Analyzer.handleSpecialFields(ua, a));
-        return potentialCandidatesIfNoMatchesFound(ua, matches, attrList, list -> UnresolvedNamePattern.errorMessage(up.pattern(), list));
+        return resolveAgainstList(
+            a -> up.match(a.name()),
+            null,
+            ua,
+            attrList,
+            true,
+            list -> UnresolvedNamePattern.errorMessage(up.pattern(), list)
+        );
     }
 
     private static List<Attribute> resolveAgainstList(UnresolvedAttribute ua, Collection<Attribute> attrList) {
-        var matches = AnalyzerRules.maybeResolveAgainstList(ua, attrList, a -> Analyzer.handleSpecialFields(ua, a));
-        return potentialCandidatesIfNoMatchesFound(ua, matches, attrList, list -> UnresolvedAttribute.errorMessage(ua.name(), list));
+        String suffix = "." + ua.name();
+        Predicate<Attribute> matchLoosely = a -> a.name().equals(ua.name()) || a.name().endsWith(suffix);
```

Comment:
This is the crux w.r.t. resolving qualified names from `ENRICH` in case it's unambiguous; e.g. if `ENRICH` added the column `a.b` and we want to just write `KEEP b` because we're lazy.

It does not distinguish between qualifiers from `ENRICH` and regular superfields.